languages/python/django-oso/django_oso/auth.py

```diff
@@ -93,11 +85,7 @@ class in Polar (``app_name::ModelName``).
         if filter is None:
             filter = Q()
 
-        next_filter = partial_to_query_filter(resource_partial, resource_type)
-        if next_filter == Q():
```

Comment:
We were early-returning on empty ANDs, which was causing the following case to fail:
```python
allow(_, _, post: test_app::Post) if check_user(post.created_by);
check_user(_: test_app::Admin);
check_user(user: test_app::Guest) if user.name = "alice";
```
`check_user(_: test_app::Admin)` added a single constraint, `_this matches test_app::Admin{}`, which evaluates to `None` in `translate_expr()`. See test case added in 01698256dbbb0214cca35d1945ba70ce3ca0ad8d

Reply:
Not sure I follow.  This case was intended to handle constraints that allow everything (an empty filter set).  Once one constraint like that is added, no others will do anything since OR of some constraint with everything is still everything.

Reply:
Added the early returning code back in c90718e